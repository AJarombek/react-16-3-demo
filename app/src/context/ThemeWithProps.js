/**
 * Implement a global theme for components using props.  This is in contrast with
 * {@link ThemeWithContext.js}, which uses the context API.
 * @author Andrew Jarombek
 * @since 1/2/2020
 */

import React, {useState} from 'react';
import { useHistory } from 'react-router-dom';
import {AJTextCard, AJSwitchIcon} from 'jarombek-react-components';
import classnames from 'classnames';

import green from '../assets/green.png';
import light from '../assets/light.png';

const ThemeWithProps = () => {
  const [theme, setTheme] = useState('light');

  return (
    <div className={classnames("theme-with-props", `${theme}-theme-with-props`)}>
      <AJSwitchIcon
        initialState={true}
        disabled={false}
        onChange={() => setTheme(theme === 'dark' ? 'light' : 'dark')}
        offImageUrl={green}
        onImageUrl={light}
      />
      <ThemeWithPropsCard theme={theme} />
    </div>
  );
};

const ThemeWithPropsCard = ({theme}) => {
  return (
    <div className={classnames('theme-with-props-card', `${theme}-theme-with-props-card`)}>
      <AJTextCard
        title="Theme implemented with props"
        content="... content ..."
        action={null}
        actionText="Button"
        actionDisabled={false}
      />
    </div>
  );
};

export default ThemeWithProps;
