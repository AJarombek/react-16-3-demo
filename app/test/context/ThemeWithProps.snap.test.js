/**
 * Snapshot test for the {@link ../../src/context/ThemeWithProps} component.
 * @author Andrew Jarombek
 * @since 1/28/2020
 */

import React from 'react';
import ThemeWithProps from '../../src/context/ThemeWithProps';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<ThemeWithProps />).toJSON();
  expect(tree).toMatchSnapshot();
});
