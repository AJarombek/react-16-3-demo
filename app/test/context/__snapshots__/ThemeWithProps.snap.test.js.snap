// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="theme-with-props light-theme-with-props"
>
  <div
    className="ajSwitchIconContainer-0-2-1 aj-switch-icon"
  >
    <figure
      className="ajSwitchIcon-0-2-2 ajSwitchIconNotSelected-0-2-3 aj-switch-icon-off"
    >
      <img
        src="test-file-stub"
      />
    </figure>
    <div
      className="ajSwitch-0-2-4 ajSwitchActive-0-2-6 aj-switch"
      onClick={[Function]}
    >
      <div
        className="ajSwitchHead-0-2-8 ajSwitchHeadActive-0-2-9 aj-switch-head"
      >
         
      </div>
      <div
        className="ajSwitchTail-0-2-11 ajSwitchTailActive-0-2-12 aj-switch-tail"
      >
         
      </div>
    </div>
    <figure
      className="ajSwitchIcon-0-2-2 aj-switch-icon-on"
    >
      <img
        src="test-file-stub"
      />
    </figure>
  </div>
  <div
    className="theme-with-props-card light-theme-with-props-card"
  >
    <div
      className="ajTextCard-0-2-14 aj-text-card"
    >
      <div
        className="aj-text-card-header"
      >
        <h2
          className="aj-text-card-title"
        >
          Theme implemented with props
        </h2>
        <p
          className="aj-text-card-subtitle"
        />
      </div>
      <div
        className="aj-text-card-body"
      >
        <p
          className="aj-text-card-content"
        >
          ... content ...
        </p>
      </div>
      <div
        className="aj-text-card-footer"
      >
        <div
          className="ajBaseButton-0-2-18 ajBaseButtonEnabled-0-2-19 aj-text-button ajTextButton-0-2-15 ajTextButtonEnabled-0-2-16"
          onClick={null}
        >
          <button
            disabled={false}
            type="button"
          >
            Button
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
